File: backend/app/schemas/store.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class StoreBase(BaseModel):
    store_code: str
    store_name: str
    address: Optional[str] = None
    phone: Optional[str] = None
    manager: Optional[str] = None
    is_active: bool = True


class StoreCreate(StoreBase):
    pass


class StoreUpdate(BaseModel):
    store_name: Optional[str] = None
    address: Optional[str] = None
    phone: Optional[str] = None
    manager: Optional[str] = None
    is_active: Optional[bool] = None


class StoreResponse(StoreBase):
    id: str = Field(alias="_id")
    created_at: datetime

    class Config:
        populate_by_name = True

File: frontend/src/api/store.ts
import request from './request'
import type { Store, PaginatedResponse } from '@/types'

export interface StoreQuery {
  page?: number
  page_size?: number
  keyword?: string
  is_active?: boolean
}

export interface StoreCreate {
  store_code: string
  store_name: string
  address?: string
  phone?: string
  manager?: string
  is_active?: boolean
}

export interface StoreUpdate {
  store_name?: string
  address?: string
  phone?: string
  manager?: string
  is_active?: boolean
}

export const storeApi = {
  list: (params?: StoreQuery) =>
    request.get<unknown, PaginatedResponse<Store>>('/stores', { params }),
  listAll: () => request.get<unknown, Store[]>('/stores/all'),
  get: (id: string) => request.get<unknown, Store>(`/stores/${id}`),
  create: (data: StoreCreate) => request.post<unknown, Store>('/stores', data),
  update: (id: string, data: StoreUpdate) => request.put<unknown, Store>(`/stores/${id}`, data),
  delete: (id: string) => request.delete<unknown, { message: string }>(`/stores/${id}`),
}
